File: aqeFrontend/src/app/hr-module/model/question.ts
export class Question {
    id: String;
    question: String;
    answer: String;
    choices: String[];
    difficulty: String;
    score: number;
    weight: number;
    
    constructor(i,q,a,c,d,s,w){
        this.id = i;this.question=q;this.answer=a;this.choices=c;this.difficulty=d;this.score=s;this.weight=w;
    }

    
}
// private String id;
// private String question;
// private String answer;
// private String[] choices;
// private Difficulty difficulty;
// private int score;
// private int weight;

File: hrService/src/main/java/com/stackroute/hrservice/model/Question.java
package com.stackroute.hrservice.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document
public class Question {
  @Id
  private String id;
  private String question;
  private String sentence;
  private String answer;
  private String[] choices;
  private String difficulty;
  private double score;
  private int correctAttempts;
  private int totalOccurrences;

  public Question() {
  }

  public Question(String id, String question, String sentence, String answer, String[] choices, String difficulty, double score, int correctAttempts, int totalOccurrences) {
    this.id = id;
    this.question = question;
    this.sentence = sentence;
    this.answer = answer;
    this.choices = choices;
    this.difficulty = difficulty;
    this.score = score;
    this.correctAttempts = correctAttempts;
    this.totalOccurrences = totalOccurrences;
  }
}
